libraries/AP_BattMonitor/AP_BattMonitor_SMBus.cpp

```diff
@@ -30,6 +30,74 @@ bool AP_BattMonitor_SMBus::get_cycle_count(uint16_t &cycles) const
     return true;
 }
 
+// return true if serial number can be provided and fills in cycles argument
+bool AP_BattMonitor_SMBus::get_serial_number(char *serial_number, uint8_t &serial_name_len) const {
+
+    if (_serial_number != -1) {
+    	snprintf(serial_number, serial_name_len, "%i", (signed) _serial_number);
+        return true;
+    }
+
+	return false;
+}
+
+// return true if device name can be provided and fills in device name
+bool AP_BattMonitor_SMBus::read_device_name()
+{
+    if (_device_name_len != 0) {
+        return true;
+    } else {
+        _device_name_len = read_block(BATTMONITOR_SMBUS_DEVICE_NAME, _device_name, true) + 1;
+
+        if (_device_name_len > 0)
+        {
+            return true;
+        }
+    }
+
+    return false;
```

Comment:
```suggestion
    return _device_name_len > 0;
```

Similarly in the like methods below